### DISCUSSION TITLE
findByXAndYOrY order of operation

### DISCUSSION THREAD

**Discussion by KarlHakanNordgren** [micronaut-projects/micronaut-data#1544] [Q&A]
Hi

I would like to match records like this: `(X == x) && (Y == y_1 || Y == y_2)`. How can this be expressed as a repository method? My attempt in the title -- `findByXAndYOrY(x, y_1, y_2)` -- results in `Cannot query entity XXX on non-existent property`.

**Comment by KarlHakanNordgren** [comment]
@dstepanov : Do you know the answer to this one?

**Reply by dstepanov** [reply]
Sorry, this is not supported, custom query would be the best solution

**Reply by KarlHakanNordgren** [reply]
Oh no. Thanks for the answer
